src/postgres_mcp/server.py

```diff
@@ -162,6 +164,84 @@ async def table_schema_resource(table_name: str) -> str:
         raise
 
 
+@mcp.tool(
+    description="Explains the execution plan for a SQL query, showing how the database will execute it and provides detailed cost estimates."
+)
+async def explain_query(
+    sql: str = Field(description="SQL query to explain"),
+    analyze: bool = Field(
+        description="When True, actually runs the query to show real execution statistics instead of estimates. Takes longer but provides more accurate information.",
+        default=False,
+    ),
+    hypothetical_indexes: list[dict[str, Any]] | None = Field(
+        description="""Optional list of hypothetical indexes to simulate. Each index must be a dictionary with these keys:
+    - 'table': The table name to add the index to (e.g., 'users')
+    - 'columns': List of column names to include in the index (e.g., ['email'] or ['last_name', 'first_name'])
+    - 'using': Optional index method (default: 'btree', other options include 'hash', 'gist', etc.)
+
+Examples: [
+    {"table": "users", "columns": ["email"], "using": "btree"},
+    {"table": "orders", "columns": ["user_id", "created_at"]}
+]""",
+        default=None,
+    ),
+) -> ResponseType:
+    """
+    Explains the execution plan for a SQL query.
+
+    Args:
+        sql: The SQL query to explain
+        analyze: When True, actually runs the query for real statistics
+        hypothetical_indexes: Optional list of indexes to simulate
+    """
+    try:
+        sql_driver = await get_sql_driver()
+        explain_tool = ExplainPlanTool(sql_driver=sql_driver)
+        result: ExplainPlanArtifact | ErrorResult | None = None
+
+        # If hypothetical indexes are specified, check for HypoPG extension
+        if hypothetical_indexes:
+            try:
```

Comment:
Check is `analyze` is true and error out due to incompatible options if so.

Reply:
Good idea. Fixed in here: https://github.com/crystaldba/postgres-mcp/pull/23/commits/c7205f17421ef57fc2bd43ac9e0a664ba0182925